test/integration/targets/ansible-galaxy-collection/tasks/list.yml

```
@@ -31,6 +31,58 @@
       - "'dev.collection2 placeholder' in list_result.stdout"
       - "'dev.collection3 *' in list_result.stdout"
 
+- name: list collections in yaml format
+  command: ansible-galaxy collection list -o yaml
+  register: list_result_yaml
+  environment:
+    ANSIBLE_COLLECTIONS_PATH: "{{ galaxy_dir }}/dev:{{ galaxy_dir }}/prod"
+
+- assert:
+    that:
+      - "item.value | length == 3"
+      - "item.value['dev.collection1'].version == '*'"
+      - "item.value['dev.collection2'].version == 'placeholder'"
+      - "item.value['dev.collection3'].version == '*'"
+  with_dict: "{{ list_result_yaml.stdout | from_yaml }}"
+
+- name: list collections in json format
+  command: ansible-galaxy collection list -o json
+  register: list_result_json
+  environment:
+    ANSIBLE_COLLECTIONS_PATH: "{{ galaxy_dir }}/dev:{{ galaxy_dir }}/prod"
+
+- assert:
+    that:
+      - "item.value | length == 3"
+      - "item.value['dev.collection1'].version == '*'"
+      - "item.value['dev.collection2'].version == 'placeholder'"
+      - "item.value['dev.collection3'].version == '*'"
+  with_dict: "{{ list_result_json.stdout | from_json }}"
+
+- name: list single collection in json format
+  command: "ansible-galaxy collection list {{ item.key }} -o json"
+  register: list_single_result_json
+  environment:
+    ANSIBLE_COLLECTIONS_PATH: "{{ galaxy_dir }}/dev:{{ galaxy_dir }}/prod"
+  with_dict: "{{ { 'dev.collection1': '*', 'dev.collection2': 'placeholder', 'dev.collection3': '*' } }}"
```

This is rather overengineered:
```suggestion
  command: "ansible-galaxy collection list {{ item }} -o json"
  register: list_single_result_json
  environment:
    ANSIBLE_COLLECTIONS_PATH: "{{ galaxy_dir }}/dev:{{ galaxy_dir }}/prod"
  loop:
  - dev.collection1
  - dev.collection2
  - dev.collection3
```

---

Sorry if it seems that way. But if you look at the assertions below, I am using the value in this dictionary to access the expected version of the collection. If you have a suggestion on how to better restructure this test, I am happy to update. But this suggestion breaks the tests.

---

@shanemcd I'm pretty it shouldn't break tests in this place. You don't use the values. The assertion below uses the results that come from the stdout output. In this specific task, you only use keys.

---

@webknjaz I am using the values in the assertion directly proceeding this task. Notice ` == item.item.value`

---

Yes, but that refers to what you feed to the assertion task's `with_dict`, nothing to do with this task.

---

Oh... I see what you did. You use a closure. This is clever but it is also hard to follow.